### PR TITLE
HLRC: Delete role API

### PR DESCRIPTION
**Pull request by ywelsch**
Adds the "Delete role" API to the high-level REST client.

### CODE REVIEW

**Comment by ywelsch** [comment]
@elasticmachine retest this please

File: client/rest-high-level/src/main/java/org/elasticsearch/client/security/DeleteRoleResponse.java [hunk 1]
```diff
@@ -0,0 +1,55 @@
...
+import org.elasticsearch.common.xcontent.XContentParser;
+
+import java.io.IOException;
+
+import static org.elasticsearch.common.xcontent.ConstructingObjectParser.constructorArg;
+
+/**
+ * Response for a role being deleted from the native realm
+ */
+public final class DeleteRoleResponse {
```
Comment (hub-cap) [inline client/rest-high-level/src/main/java/org/elasticsearch/client/security/DeleteRoleResponse.java]: We have some better tests to test the responses now too. Check out https://github.com/elastic/elasticsearch/blob/master/client/rest-high-level/src/test/java/org/elasticsearch/client/watcher/ActivateWatchResponseTests.java

**Review by hub-cap — CHANGES_REQUESTED** [review]
Mostly nits and missing one test. Easy peasy!

File: docs/java-rest/high-level/security/delete-role.asciidoc [hunk 2]
```diff
@@ -0,0 +1,46 @@
+[[java-rest-high-security-delete-role]]
```
[the remark this answers is not in the document]
Comment (ywelsch) [reply docs/java-rest/high-level/security/delete-role.asciidoc]: fixed in 86cf213

File: client/rest-high-level/src/main/java/org/elasticsearch/client/security/DeleteRoleResponse.java (hunk 1 above)
Comment (ywelsch) [reply client/rest-high-level/src/main/java/org/elasticsearch/client/security/DeleteRoleResponse.java]: fixed in 
c041014

File: client/rest-high-level/src/main/java/org/elasticsearch/client/security/DeleteRoleRequest.java [hunk 3]
```diff
@@ -0,0 +1,57 @@
...
+    public String getName() {
+        return name;
+    }
+
+    public RefreshPolicy getRefreshPolicy() {
+        return refreshPolicy;
+    }
+
+    @Override
+    public XContentBuilder toXContent(XContentBuilder builder, Params params) {
```
[the remark this answers is not in the document]
Comment (ywelsch) [reply client/rest-high-level/src/main/java/org/elasticsearch/client/security/DeleteRoleRequest.java]: fixed in 86cf213

File: docs/java-rest/high-level/security/delete-role.asciidoc (hunk 2 above)
[the remark this answers is not in the document]
Comment (ywelsch) [reply docs/java-rest/high-level/security/delete-role.asciidoc]: fixed in 27590cf

**Review by hub-cap — APPROVED** [review]

**Pull request merged**